frontend/src/datasources/LeagueDataSource.ts

```diff
@@ -0,0 +1,47 @@
+
+export class LeagueDataSource {
+
+    baseURL: string = "http://localhost:3001/api/league";
+
+    public async createLeague(leagueName: string): Promise<
+    { error: string ; league?  : {id: string, name: string}}
+    > {
+        const response: Response = await fetch(`${this.baseURL}/create`, {
+            method: "POST",
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            credentials: "include",
+            body: JSON.stringify({name : leagueName})
+        });
+
+        if (!response.ok) {
+            const error = await response.json();
+            return { error: error.error || 'Error in league creation'}
+        }
+        
+        const league = await response.json();
+        return { error: "", league};
+    }
```

Comment:
dans notre cas, on a pas besoin de la ligue qui est retourné une fois créée donc on return error: "" (empty string, empty error -> no error)

```suggestion
    public async createLeague(leagueName: string): Promise<{ error: string }> {
        const response: Response = await fetch(`${this.baseURL}/create`, {
            method: "POST",
            headers: {
                'Content-Type': 'application/json',
            },
            credentials: "include",
            body: JSON.stringify({ name: leagueName })
        });

        if (!response.ok) {
            const error = await response.json();
            return { error: error.error || 'Error in league creation' }
        }

        return { error: "" };
    }
```